pulp_smash/tests/ostree/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,447 @@
+            {'criteria': {}})
+            for repo in repos
+        ))
+
+        cls.units = []
+        cls.original_units = []
+        # download files from server over https
+        for branch in _VALID_BRANCHES:
+            for distributor in cls.responses['distribute']:
+                url = urljoin(
+                    cls.cfg.base_url,
+                    'pulp/ostree/web/' +
+                    distributor.json()['config']['relative_path'] +
+                    '/refs/heads/' +
+                    branch)
+                response = client.get(url)
+                cls.units.append(response.content)
+                url_orig = urljoin(_VALID_FEED, 'refs/heads/' + branch)
+                response = client.get(url_orig)
+                cls.original_units.append(response.content)
+                cls.original_units[-1] = cls.original_units[-1].replace('\n', '')  # noqa pylint:disable=line-too-long
+
+        for repo in repos:
+            cls.resources.add(repo['_href'])
+
+    def test_start_sync_code(self):
+        """Assert the call to sync a repository returns an HTTP 202."""
+        steps_codes = (
+            ('update', 200),
+            ('sync', 202),
+            ('distribute', 201),
+            ('publish', 202),
+            ('copy', 202),
+            ('search units', 200),
+        )
+        for step, code in steps_codes:
+            with self.subTest((step, code)):
+                for response in self.responses[step]:
+                    self.assertEqual(response.status_code, code)
+
+    def test_task_error(self):
+        """Assert each task's "error" field is null."""
+        for step in {'sync', 'copy'}:
+            for i, task_body in enumerate(self.task_bodies[step]):
+                with self.subTest(i=i):
+                    self.assertEqual(task_body['error'], None)
+
+    def test_task_traceback(self):
+        """Assert each task's "traceback" field is null."""
+        for step in {'sync', 'copy'}:
+            for i, task_body in enumerate(self.task_bodies[step]):
+                with self.subTest(i=i):
+                    self.assertEqual(task_body['traceback'], None)
+
+    def test_task_progress_report(self):
+        """Assert no task's progress report contains error details."""
+        for i, task_body in enumerate(self.task_bodies['sync']):
+            for action in task_body['progress_report']['ostree_web_importer']:
+                with self.subTest(i=i):
+                    self.assertEqual(
+                        len(action['error_details']),
+                        0
+                    )
+
+    def test_search_units(self):
+        """Verify that the content of two repositories differ."""
+        repo0set = set(unit['unit_id']
+                       for unit in self.responses['search units'][0].json())
+        repo1set = set(unit['unit_id']
+                       for unit in self.responses['search units'][1].json())
```

Comment:
Please avoid hanging indents. It's purely stylistic, but they're used nowhere else in Pulp Smash.